tooling/bundler/src/bundle/linux/pacman.rs

```diff
@@ -0,0 +1,241 @@
+    pkgrel,
+    arch
+  );
+
+  let package_name = format!("{}.pkg.tar.zst", package_base_name);
+
+  let package_dir = settings.project_out_directory().join("bundle/pacman");
+  if package_dir.exists() {
+    fs::remove_dir_all(&package_dir)
+      .with_context(|| format!("Failed to remove old {}", package_base_name))?;
+  }
+  let package_path = package_dir.join(&package_name);
+
+  info!(action = "Bundling"; "{} ({})", package_name, package_path.display());
+
+  copy_bin_lib(settings, &package_dir).with_context(|| "Failed to copy lib and bin file")?;
+  generate_desktop_file(settings, &package_dir).with_context(|| "Failed to create desktop file")?;
+  generate_icon_files(settings, &package_dir).with_context(|| "Failed to create icon file")?;
+  generate_pkgbuild(settings, &package_dir, arch, pkgrel).with_context(|| "Failed to create PKGBUILD file")?;
+  run_pkgbuild(&package_dir).with_context(|| "Failed to run PKGBUILD file")?;
+
+  Ok(vec![package_path])
+}
+
+fn copy_bin_lib(settings: &Settings, package_dir: &Path) -> crate::Result<()> {
+  let bin_dir = package_dir.join("usr/bin");
+  let resource_dir = package_dir.join("usr/lib").join(settings.main_binary_name());
+
+  for bin in settings.binaries() {
+    let bin_path = settings.binary_path(bin);
+    common::copy_file(&bin_path, &bin_dir.join(bin.name()))
+      .with_context(|| format!("Failed to copy binary from {:?}", bin_path))?;
+  }
+
+  settings
+    .copy_binaries(&bin_dir)
+    .with_context(|| "Failed to copy external binaries")?;
+  settings.copy_resources(&resource_dir).with_context(|| "Failed to copy resource files")?;
+
+  Ok(())
+}
+
+/// Generate the application desktop file and store it under the `data_dir`.
+fn generate_desktop_file(settings: &Settings, package_dir: &Path) -> crate::Result<()> {
+  let bin_name = settings.main_binary_name();
+  let desktop_file_name = format!("{}.desktop", bin_name);
+  let desktop_file_path = package_dir
+    .join("usr/share/applications")
+    .join(desktop_file_name);
+  let file = &mut common::create_file(&desktop_file_path)?;
+  // For more information about the format of this file, see
+  // https://developer.gnome.org/integration-guide/stable/desktop-files.html.en
+  writeln!(file, "[Desktop Entry]")?;
+  if let Some(category) = settings.app_category() {
+    writeln!(file, "Categories={}", category.gnome_desktop_categories())?;
+  } else {
+    writeln!(file, "Categories=")?;
+  }
+  if !settings.short_description().is_empty() {
+    writeln!(file, "Comment={}", settings.short_description())?;
+  }
+  writeln!(file, "Exec={}", bin_name)?;
+  writeln!(file, "Icon={}", bin_name)?;
+  writeln!(file, "Name={}", settings.product_name())?;
+  writeln!(file, "Terminal=false")?;
+  writeln!(file, "Type=Application")?;
+  Ok(())
+}
+
+
+/// Generate the icon files and store them under the `data_dir`.
+fn generate_icon_files(settings: &Settings, package_dir: &Path) -> crate::Result<BTreeSet<LinuxIcon>> {
```

Comment:
This looks identical to the debian version (perhaps other functions too?). Could you break out any shared code, perhaps into a lib file in the parent (linux) module?

(Note https://github.com/tauri-apps/tauri/pull/4282 which changes this logic)